src/main/java/executor/service/executionService/ParallelFlowExecutorService.java

```diff
@@ -0,0 +1,76 @@
+package executor.service.executionService;
+
+import executor.service.executor.ExecutionService;
+import executor.service.model.ThreadPoolConfig;
+import executor.service.scenario.ScenarioExecutor;
+import executor.service.scenario.ScenarioSourceListenerImpl;
+import executor.service.utils.PropertiesConfigHolder;
+import executor.service.webdriver.WebDriverInitializer;
+
+import org.openqa.selenium.WebDriver;
+
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class ParallelFlowExecutorService {
+
+    private final ThreadPoolConfig threadPoolConfig;
+    private ThreadPoolExecutor threadPoolExecutor;
+    private final ScenarioSourceListenerImpl scenarioSourceListener;
+    private final ExecutionService executionService;
+    private final WebDriver webDriver;
```

Comment:
This field might need to be WebDriverInitializer, since each separate thread will need its own driver to work with, and initializer can provide them